pipeline-runner/R/gem2s-2-load_cellranger_files.R

```diff
@@ -98,3 +99,66 @@ format_annot <- function(annot_list) {
   rownames(annot) <- annot$input
   return(annot)
 }
+
+
+read_rhapsody_matrix <- function(config, input_dir) {
+  counts_list <- list()
+  annot_list <- list()
+
+  samples <- config$samples
+  message("Samples to include in the analysis:\n- ", paste(samples, collapse = "\n- "))
+  message("Loading rhapsody data set from input folder.")
+
+
+  for (sample in samples) {
+    sample_dir <- file.path(input_dir, sample)
+    sample_fpaths <- file.path(sample_dir, "expression_matrix.st")
+
+    message("\nSample --> ", sample)
+    message("Reading files from ", sample_dir, " --> ", paste(sample_fpaths, collapse = " - "))
+
+    counts <- data.table::fread(sample_fpaths)
+
+    # catch absent DBEC column
+    adjusted_col <- ifelse(
+      "DBEC_Adjusted_Molecules" %in% colnames(counts),
+      "DBEC_Adjusted_Molecules",
+      "RSEC_Adjusted_Molecules"
+    )
+
+    # The ..keep is data.table syntax to grab the keep columns
+    keep <- c("Cell_Index", "Gene", adjusted_col)
+    counts <- counts[, ..keep]
+
+    counts[, Gene := factor(Gene)]
+    counts[, gene_i := as.integer(Gene)]
+
+    features <- levels(counts$Gene)
+    barcodes <- unique(counts$Cell_Index)
+
+    # to create small sparse matrix, and retain original cell indices ("barcodes")
+    counts[, cell_index_j := match(Cell_Index, barcodes)]
```

Comment:
I would consider removing all the `data.table` syntax (other than fread) here as it's not necessary and may be unfamiliar to future developers (or current).